### ISSUE TITLE
Document the use of HTML elements in System.Web.UI.WebControls.TreeNode.Text

### ISSUE THREAD

**Issue by mikebk** [MicrosoftDocs/feedback#326]
# Idea Title

 Document the use of HTML elements in System.Web.UI.WebControls.TreeNode.Text

## Issue Details

The System.Web.UI.WebControls.TreeNode.Text property accepts HTML elements but there appears to be no documentation on the effects of including these html fragments.

There are two problems with this: First, the documentation doesn't include the normal warning about cross Site Scripting vulnerabilities (you'll notice the example below includes some HtmlEncoding). Second, there's no description of how to adjust the visual appearance of the text.

As the property doesn't expose any methods/properties to adjust the visual appearance of the text
there have been quite a number of developers posting inquiries (e.g. <https://forums.asp.net/t/1342537.aspx?Change+TreeView+treenode+background+color> or <https://forums.asp.net/t/1342537.aspx?Change+TreeView+treenode+background+color>  about how to set the color of the text. The typical solution is along the lines of 'childNode.Text = "&lt;div style='color:Red'>" + childNode.Text + "&lt;/div>";"

Mike

**Comment by Powerhelmsman** [comment]
@mikebk Can you provide the document link here?

**Comment by dend** [comment]
@Rick-Anderson @danroth27 can you please help with this ask?

**Comment by Rick-Anderson** [comment]
@mikebk  is [this the doc](https://msdn.microsoft.com/en-us/library/system.web.ui.webcontrols.treenode.text(v=vs.110).aspx)

@dend  I'm not familiar with `System.Web` but @danroth27 might know who is.

**Comment by Powerhelmsman** [comment]
@danroth27 Can you response if you are the right person to contact?

**Comment by csharpfritz** [comment]
Adding @HongGit to review and schedule

**Comment by HongGit** [comment]
@mikebk if you browse a bit on the MSDN documentation, you will find the [Working with ASP.NET Web Server Controls](https://msdn.microsoft.com/en-us/library/k26w4zzx(v=vs.110).aspx) page, where you will find two of the topics you are querying about:
- [How to: Set ASP.NET Web Server Control Properties](https://msdn.microsoft.com/en-us/library/7ataet98.aspx), which does have Security Note regarding Script Exploits Overview.
- [ASP.NET Web Server Controls and CSS Styles](https://msdn.microsoft.com/en-us/library/h4kete56.aspx) might be of interest to you, which talks about how tp control ASP.NET server controls by setting various appearance properties such as ForeColor, etc.

Hope this is helpful!

**Comment by Powerhelmsman** [comment]
@mikebk I will close this issue now. If you have other question related to it, we will reopen the issue and continue the discussion.

**Comment by mikebk** [comment]
I must apologize for not responding to the earlier messages in this conversation. For some reason, I wasn't notified they had been posted. I've tried to adjust my email settings and I hope I receive responses now. I had expected that the feedback collection software would capture the URL of the document I was commenting on.

I created the issue after investigating a security vulnerability reported by IBM's AppScan program. I thought I would take advantage of the Feedback facility to improve the documentation for future developers. 

There were two improvements that I thought would help this First, to improve security, include in the TreeNode document a warning about user input like that provided for other controls (e.g., https://msdn.microsoft.com/en-us/library/system.web.ui.webcontrols.label(v=vs.110).aspx) to make it more obvious to the developer (and as some controls do HtmlEncode text).

Second, while analyzing the code our developer had written, I realized I wasn't sure what the inline HTML was doing. I found quite a number developers asking how to format the text in the control. That's why I suggested some text on formatting the contents of the TreeNode. I do agree that css is a better way to set the formatting but apparently, that wasn't obvious to the developers looking for assistance.
